app/views/campaign_pages/edit.slim

```diff
@@ -0,0 +1,13 @@
+= form_for(@campaign_page, class: 'form-horizontal') do |f|
+  h2 = @campaign_page.title
+
+  div.form-group
+    = f.label :title, class: 'control-label'
+    = f.text_field :title, class: 'col-lg-3', placeholder: @campaign_page.title
+
+  - @campaign_page.campaign_pages_widgets.each do |widget|
```

Comment:
When I went to do an edit on a petition page I'd created, I was getting an error because the forms didn't have the `widget_type_id` that they were expecting defined. Did you run into that at all?